zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestRawZkClient.java

```diff
@@ -913,6 +913,79 @@ public void testRetryUntilConnectedAfterConnectionLoss()
     _zkClient.delete(path);
   }
 
+  /*
+   * Test the scenario where we have actually lost connection to ZK but
+   * the internal client connection state assumes that we are still connected.
+   * Once we re-start the ZK, the creation operation will finish.
+   * How this test simulates the steps is:
+   * 1. Shuts down zk server
+   * 2. Starts a creation thread to create an ephemeral node in the original zk session
+   * 3. Creation operation loses connection and keeps retrying
+   * 4. Restarts zk server and Zk session is recovered
+   * 5. zk client reconnects successfully and creates an ephemeral node
+   */
+  @Test(timeOut = 5 * 60 * 1000L)
```

Comment:
Question: do we want to put this test in test sequence? (probably the last one?)

Reply:
thanks for the suggestion, let me try that.